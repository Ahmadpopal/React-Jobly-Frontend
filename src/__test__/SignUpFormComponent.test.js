import React from 'react';
import UserContext from '../Context';
import { render } from '@testing-library/react';
import { MemoryRouter} from 'react-router-dom';
import SignUp from '../SignUpFormComponent'


describe('SIGN UP TESTS', () => {

    // SMOKE TEST 
    it('SignUp Component should render without crushing', ()=>{
        let currentUser = null;
        render(
            <UserContext.Provider value={{ currentUser }} >
                <MemoryRouter>
                    <SignUp />
                </MemoryRouter>
            </UserContext.Provider>
        )
    }) 

    // SNAPSHOT TEST 
    it('snapshot should mataches', () => {
        let currentUser = null;
        const { asFragment } = render(
            <UserContext.Provider value={{ currentUser }} >
            <MemoryRouter>
                <SignUp />
            </MemoryRouter>
        </UserContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    })


    it('shows a signUp form', () => {
		let currentUser = null;
		const { getByText } = render(
			<UserContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<SignUp />
				</MemoryRouter>
			</UserContext.Provider>
		);
		expect(getByText('Register')).toBeInTheDocument();
	});

});