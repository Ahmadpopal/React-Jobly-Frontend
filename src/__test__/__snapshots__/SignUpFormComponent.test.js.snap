// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGN UP TESTS snapshot should mataches 1`] = `
<DocumentFragment>
  <div
    class="SignUp"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="SignUp-div-title col"
        >
          <h3
            class="SingUp-h3"
          >
            Sign Up
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <form
            class="SignUp-form"
          >
            <div
              class="form-group"
            >
              <label
                class="SignUp-label"
                for="username"
              >
                Username
              </label>
              <input
                class="SignUp-input form-control"
                name="username"
                placeholder="Username here"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="SignUp-label"
                for="password"
              >
                Password
              </label>
              <input
                class="SignUp-input form-control"
                name="password"
                placeholder="Username here"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="SignUp-label"
                for="firstName"
              >
                First Name
              </label>
              <input
                class="SignUp-input form-control"
                name="firstName"
                placeholder="Username here"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="SignUp-label"
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="SignUp-input form-control"
                name="lastName"
                placeholder="Username here"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="SignUp-label"
                for="email"
              >
                Email
              </label>
              <input
                class="SignUp-input form-control"
                name="email"
                placeholder="Username here"
                type="email"
                value=""
              />
            </div>
            <button
              class="SignUp-button btn btn-primary"
            >
              Register
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
